test/v5/kyberBancorReserve.js

```diff
@@ -0,0 +1,576 @@
+        }
+
+        try {
+            await reserve.trade(ethAddress, amountEth, bancorBntToken.address, user, ethToBntRate, true, {from: network, value: amountEth.sub(10)});
+            assert(false, "throw was expected in line above.")
+        } catch (e) {
+            assert(Helper.isRevertErrorMessage(e), "expected throw but got: " + e);
+        }
+
+        try {
+            await reserve.trade(bancorBntToken.address, amountBnt, ethAddress, user, bntToEthRate, true, {from: network, value: 1});
+            assert(false, "throw was expected in line above.")
+        } catch (e) {
+            assert(Helper.isRevertErrorMessage(e), "expected throw but got: " + e);
+        }
+    });
+
+    it("Should test trade conversion rate is 0", async function() {
+        let amountEth = (new BigNumber(10)).pow(18);
+        let amountBnt = (new BigNumber(10)).pow(18).mul(500);
+
+        try {
+            await reserve.trade(ethAddress, amountEth, bancorBntToken.address, user, 0, true, {from: network, value: amountEth});
+            assert(false, "throw was expected in line above.")
+        } catch (e) {
+            assert(Helper.isRevertErrorMessage(e), "expected throw but got: " + e);
+        }
+
+        try {
+            // bypass validating but user expected amount is still 0
+            await reserve.trade(ethAddress, amountEth, bancorBntToken.address, user, 0, false, {from: network, value: amountEth});
+            assert(false, "throw was expected in line above.")
+        } catch (e) {
+            assert(Helper.isRevertErrorMessage(e), "expected throw but got: " + e);
+        }
+
+        try {
+            await reserve.trade(bancorBntToken.address, amountBnt, ethAddress, user, 0, true, {from: network});
+            assert(false, "throw was expected in line above.")
+        } catch (e) {
+            assert(Helper.isRevertErrorMessage(e), "expected throw but got: " + e);
+        }
+
+        try {
+            // bypass validating but user expected amount is still 0
+            await reserve.trade(bancorBntToken.address, amountBnt, ethAddress, user, 0, true, {from: network});
+            assert(false, "throw was expected in line above.")
+        } catch (e) {
+            assert(Helper.isRevertErrorMessage(e), "expected throw but got: " + e);
+        }
+    });
+
+    it("Should test trade is reverted when conversion rate is higher than actual rate", async function() {
+        let amountEth = (new BigNumber(10)).pow(18);
+        let amountBnt = (new BigNumber(10)).pow(18).mul(500);
+
+        await reserve.setFeeBps(0, {from: admin});
+
+        let ethToBntRate = await reserve.getConversionRate(ethAddress, bancorBntToken.address, amountEth, 0);
+        let bntToEthRate = await reserve.getConversionRate(bancorBntToken.address, ethAddress, amountBnt, 0);
+
+        try {
+            await reserve.trade(ethAddress, amountEth, bancorBntToken.address, user, ethToBntRate.add(2), true, {from: network, value: amountEth});
```

Comment:
failing on Bancor contract.
right?
is this actual Bancor behavior?

Reply:
Should be our logic, we get dest amount from bancor and compare with our expected dest amount